starlite/utils/exception.py

```diff
@@ -0,0 +1,30 @@
+from inspect import isclass
+from typing import Dict, Optional, Type, Union
+
+from starlette.exceptions import HTTPException
+from starlette.status import HTTP_500_INTERNAL_SERVER_ERROR
+
+from starlite.types import ExceptionHandler
+
+
+def get_exception_handler(
+    exception_handlers: Dict[Union[int, Type[Exception]], ExceptionHandler], exc: Exception
+) -> Optional[ExceptionHandler]:
+    """
+    Given a dictionary that maps exceptions and status codes to handler functions,
+    and an exception, returns the appropriate handler if existing.
+    """
+    if not exception_handlers:
+        return None
+    if isinstance(exc, HTTPException) and exc.status_code in exception_handlers:
+        return exception_handlers[exc.status_code]
+    if exc.__class__ in exception_handlers:
+        return exception_handlers[exc.__class__]
+    if HTTP_500_INTERNAL_SERVER_ERROR in exception_handlers:
+        return exception_handlers[HTTP_500_INTERNAL_SERVER_ERROR]
+    for key, value in exception_handlers.items():
```

Comment:
Another way to do this could be to walk the mro of the the exception type and check for an entry in `exception_handlers` for the class and each of it's bases, e.g., 

```python
def get_exception_handler(...) -> ...:
    if not exception_handlers:
        return None
    if isinstance(exc, HTTPException) and exc.status_code in exception_handlers:
        return exception_handlers[exc.status_code]
    for cls in type(exc).__mro__:
        if cls not in exception_handlers:
            continue
        return exception_handlers[cls]
    return None
```
I haven't tested though.